magma/debug.py

```diff
@@ -1,43 +1,43 @@
-import inspect
+from dataclasses import dataclass
+import uinspect
 import collections
-import magma
-from magma.config import get_debug_mode
 
+from magma.config import get_debug_mode, config
 
-debug_info = collections.namedtuple("debug_info", ["filename", "lineno", "module"])
 
+@dataclass
+class _DebugInfo:
+    filename: str
+    lineno: int
+    module: str
 
-def get_callee_frame_info():
-    callee_frame = inspect.currentframe()
-    # FIXME: Right now we assume a max 10 frames deep
-    for i in range(0, 10):
-        module = inspect.getmodule(callee_frame)
-        # Go up until we're out of the magma module (assuming this is the user
-        # code)
-        if not module or module.__name__.split(".")[0] != "magma":
-            break
-        callee_frame = callee_frame.f_back
-    callee_frame = inspect.getframeinfo(callee_frame)
-    return debug_info(callee_frame.filename, callee_frame.lineno, module)
+
+debug_info = _DebugInfo
+
+
+def get_debug_info(frames_to_skip):
+    if config.use_uinspect:
```

Comment:
let's actually register this config in this file locally, rather than in the global config.